File: collaborators/AdamVinueza/recursion_exercise/runTests.js
#!/usr/bin/env node
/*
 * Just gets the lengths of increasingly large arrays of numbers. It shouldn't
 * matter what's in the arrays, because getLength doesn't ever look inside
 * the array, but if you want to test that, feel free.
 */
const recursionExercise = require('./recursionExercise.js');

for (let expected = 0; expected < 1000; expected++) {
  const actual = recursionExercise.getLength(new Array(expected));
  if (expected != actual) {
    console.log(`Error: expected ${expected}, actual ${actual}`);
  }
}

File: Contributors/AdamVinueza/recursion_exercise/run_tests.py
#!/usr/bin/python

import recursion_exercise
import sys

def get_list_of_size(size):
    return list(range(size))

def main():
    '''
    Just gets the lengths of increasingly large lists of numbers. It shouldn't
    matter what's in the lists, because get_length doesn't ever look inside
    the list, but if you want to test that, feel free.

    Because we're testing a recursive function, we'll have to keep an eye on
    the depth of the call stack.
    '''
    sys.setrecursionlimit(2000) # Python's default limit is 1000
    empty = []
    for expected in range(1000):
        lst = get_list_of_size(expected)
        actual = recursion_exercise.get_length(lst)
        if (expected != actual):
            print('Error: expected {0}, actual {1}'.format(expected, actual))

if __name__ == '__main__':
    main()
